src/main/java/org/gelecekbilimde/scienceplatform/user/model/entity/UserEntity.java

```diff
@@ -82,8 +80,20 @@ public class UserEntity extends BaseEntity {
 	private RoleEntity roleEntity;
 
 	@ManyToMany
-	@JoinTable(name = "user_followers", joinColumns = @JoinColumn(name = "follower_user_id"), inverseJoinColumns = @JoinColumn(name = "followed_user_id"))
-	private Set<UserEntity> followerUserEntities = new HashSet<>();
+	@JoinTable(
+		name = "user_followers",
+		joinColumns = @JoinColumn(name = "follower_user_id"),
+		inverseJoinColumns = @JoinColumn(name = "followed_user_id")
+	)
+	private List<UserEntity> followings = new ArrayList<>();
+
+	@ManyToMany
+	@JoinTable(
+		name = "user_followers",
+		joinColumns = @JoinColumn(name = "followed_user_id"),
+		inverseJoinColumns = @JoinColumn(name = "follower_user_id")
+	)
+	private List<UserEntity> followers = new ArrayList<>();
```

Comment:
Set, Arraylist'e oranla performans anlamında daha iyi sonuç veriyor diye biliyorum. @agitrubard ile bunu daha önce konuşmuştuk. Arraylist olarak kalmasının özel bir nedeni yoksa performans nedeniyle Set kalabilir.

Reply:
Set daha iyi olabilir fakat mapstruct'ta Set'le ilgili bir problem var diye biliyorum, onu manuel olarak çözmemiz gerekecek, özel bir konfigürasyon yapmayalım şu an için. İlerleyen dönemde buraya dönüp bakabiliriz performans problemleri yaşadığımız zamanlarda.